app/helpers/application_helper.rb

```diff
@@ -650,4 +650,10 @@ def permit_all_except(params, blocked_keys)
       params
     end
   end
+
+  def disallow_robots?(user, creation)
```

Comment:
I don't think we need two arguments here, since we're never going to have both a user and a creation at the same time. We can also use Ruby's safe operator (`&.`) instead of Rails's `try` method:

```ruby
  def disallow_robots?(item)
    return unless item

    if item.is_a?(User)
      item&.preference&.minimize_search_engines?
    else
      item.users.all? { |u| u&.preference&.minimize_search_engines? }
    end
  end
 ```
 
 Then in the view, we'd do:
 ```ruby
<% item = @user || @work || @series %>
<% if disallow_robots?(item) %>
```